### ISSUE TITLE
[Bug]: Toasts are not announced by TalkBack screen reader

### ISSUE THREAD

**Issue by mary-ext** [radix-vue/radix-vue#2469]
### Environment

```bash
Developement/Production OS: Arch Linux
Node version: 22.20.0
Package manager: pnpm@10.13.1
Reka UI version: 2.6.1 (also reproducible on 2.8.2)
Vue version: 3.5.28
Client OS: Android 16; CPH2651 Build/BP2A.250605.015
Browser: Chrome 145.0.7632.109
```

### Link to minimal reproduction

https://stackblitz.com/edit/vitejs-vite-6j4mvhev

### Steps to reproduce

1. Open the reproduction (or https://reka-ui.com/docs/components/toast)
2. Click the button

### Describe the bug

TalkBack does not read toasts out loud

### Expected behavior

It should read toasts out loud

### Context & Screenshots (if applicable)

A customer reports it also doesn't alert on VoiceOver but I don't have an iOS/macOS device to test this personally

**Comment by mary-ext** [comment]
my guess is that it's because we aren't passing the prop that'd unset the aria-hidden in VisuallyHidden, but not sure

https://github.com/unovue/reka-ui/blob/8c6631be275cca595c334cf445d3053b5bd0ad4b/packages/core/src/Toast/ToastAnnounce.vue#L19

EDIT: setting the correct mode did not fix it
